pyquil/api/_qpu.py

```diff
@@ -76,7 +76,14 @@ def __init__(self, endpoint: str, user: str = "pyquil-user") -> None:
         :param user: A string identifying who's running jobs.
         """
         super().__init__()
-        self.client = Client(endpoint)
+        try:
+            self.client = Client(endpoint)
+        except TypeError:
+            raise Exception("""It looks like you've tried to run a program against a QPU, but do not currently have a reservation on one. To reserve time on Rigetti QPUs, use the command line interface, qcs, which comes pre-installed in your QMI. From within your QMI, type:
```

Comment:
I also think this error occurs because endpoint is None, which we should be able to explicitly check for rather than the try/except

Reply:
`RuntimeError` would be appropriate here if you want to avoid defining your own exception type

Reply:
+1 to RuntimeError, feels right. also +1 to checking for None, especially since that is the case we are writing this message for (rather than say, endpoint=1 or something silly like that)

Reply:
also this needs some line breaks